### PR TITLE
(fleet) downloader flavors

### PR DESCRIPTION
**Pull request by arbll**
### What does this PR do?

This PR adds supports for "flavors" to the installer downloader:
- Downloaders are pre-built with a flavor and version of the installer to download
- They download the version and execute `datadog-installer setup --flavor <flavor>`
- The installer runs the setup logic for the flavor 

I had to move some packages around to make imports work.

### Describe how you validated your changes
- Not used in production yet, validation will come later when we plug this in e2e.

### CODE REVIEW

**Review by raphaelgavache — APPROVED** [review]

File: cmd/installer-downloader/main.go [hunk 1]
```diff
@@ -7,20 +7,94 @@
...
 )
 
 func main() {
+	if Version == "" || Flavor == "" {
+		fmt.Fprintln(os.Stderr, "Version and Flavor must be set at build time.")
+		os.Exit(1)
+	}
 	if !user.IsRoot() {
-		fmt.Fprintln(os.Stderr, "This command requires root privileges.")
+		fmt.Fprintln(os.Stderr, "This installer requires root privileges.")
```
Comment (BaptisteFoy) [inline cmd/installer-downloader/main.go:41]: isn't it the downloader? :trollface:

**Review by BaptisteFoy — APPROVED** [review]

File: cmd/installer-downloader/main.go [hunk 2]
```diff
@@ -7,20 +7,94 @@
...
 		os.Exit(1)
 	}
-	cmd := installer.BootstrapCommand()
-	cmd.SilenceUsage = true
-	os.Exit(runcmd.Run(cmd))
+	env := env.FromEnv()
+	ctx := context.Background()
+
+	t := telemetry.NewTelemetry(env.HTTPClient(), env.APIKey, env.Site, "datadog-installer-downloader")
```
Comment (raphaelgavache) [inline cmd/installer-downloader/main.go:47]: I think flavor should be in the service name

File: cmd/installer-downloader/main.go (hunk 1 above)
Comment (arbll) [reply cmd/installer-downloader/main.go:41]: not from the customer pov haha

**Review by amenasria — APPROVED** [review]
LGTM for Agent infra owned files

**Review by FlorentClarret — APPROVED** [review]

**Pull request merged**